bot/controllers/welcome.js

```diff
@@ -0,0 +1,34 @@
+const BaseController = require('../baseController.js');
+const Command = require('../baseCommand.js');
+const util = require('apex-util');
+
+class WelcomeController extends BaseController {
+  constructor(message) {
+    // Call BaseController constructor
+    super(message);
+
+    // Aliasing 'this' as controller to allow for binding in actions
+    const controller = this;
+
+    // Array of all commands, see baseCommand.js for prototype
+    this.commands = [
+      new Command(
+        '!welcome',
+        '!welcome',
+        'Send Welcome Message',
+        'Sends A Direct Welcome Message.',
```

Comment:
Sends a direct welcome message is slightly ambiguous, after all, it is sending "the" welcome message, not "a" welcome message. And more specifically, it is resending the welcome message to the initiating user.